.gitlab-ci.yml

```diff
@@ -0,0 +1,220 @@
+    - echo "=== Running before script ==="
+    - git checkout $CI_COMMIT_BRANCH
+    - mkdir -p -m 777 $ARTIFACTS_FOLDER/
+    - |
+      if [[ -f "$BASH_ENV" ]]; then
+        source "$BASH_ENV"
+      fi
+    - source .circleci/content_release_vars.sh
+    - echo "=== Creating new clean logs folder ==="
+    - rm -rf $ARTIFACTS_FOLDER/logs
+    - mkdir -p $ARTIFACTS_FOLDER/logs
+    - echo "=== Granting execute permissions on files ==="
+    - chmod +x ./Tests/scripts/*
+    - chmod +x ./Tests/Marketplace/*
+    - echo "=== Installing Virtualenv ==="
+    - |
+     if [ -f "./venv/bin/activate" ]; then
+       echo "found venv"
+     else
+      echo "installing venv"
+      NO_HOOKS=1 SETUP_PY2=yes .hooks/bootstrap >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+     fi
+     venv/bin/pip3 install -r .circleci/build-requirements.txt >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+     venv/bin/pip2 install -r dev-requirements-py2.txt >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+     source ./venv/bin/activate
+    - git config diff.renameLimit 6000
+    - export PATH="${PWD}/node_modules/.bin:${PATH}"
+    - |
+      if [ -n "${DEMISTO_SDK_NIGHTLY}" ]; then
+        echo "Installing SDK" >> $ARTIFACTS_FOLDER/logs/installations.log
+        pip3 install git+https://github.com/demisto/demisto-sdk@master >> $ARTIFACTS_FOLDER/logs/installations.log 2>&1
+      fi
+    - echo "========= Installing SSH keys ========"
+    - 'command -v ssh-agent >/dev/null || ( apt-get update -y && apt-get install openssh-client -y )'
+    - eval $(ssh-agent -s)
+    - chmod 400 $OREGON_CI_KEY
+    - ssh-add $OREGON_CI_KEY
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+
+    - echo "======= Download configuration ========"
+    - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+
+    - echo "========== Build Parameters =========="
+    - set | grep -E "^NIGHTLY=|^INSTANCE_TESTS=|^SERVER_BRANCH_NAME=|^ARTIFACT_BUILD_NUM=|^DEMISTO_SDK_NIGHTLY=|^TIME_TO_LIVE=|^CONTRIB_BRANCH=|^FORCE_PACK_UPLOAD=|^PACKS_TO_UPLOAD=|^BUCKET_UPLOAD=|^GCS_MARKET_BUCKET=|^SLACK_CHANNEL="
+    - python --version
+    - python3 --version
+    - node --version
+    - npm --version
+    - demisto-sdk --version
+
+
+variables:
+  DONT_CACHE_LAST_RESPONSE: "true"
+  GCS_MARKET_BUCKET: "marketplace-dist"
+  SLACK_CHANNEL: "dmst-content-team"
+  DEMISTO_README_VALIDATION: "true"
+  ARTIFACTS_FOLDER: "/builds/xsoar/content/artifacts"
+  BASH_ENV: "/builds/xsoar/content/artifacts/bash_env"
+  PYTHONPATH: "/builds/xsoar/content"
+  FEATURE_BRANCH_NAME: "v4.5.0"
+# Gitlab CI/CD uses shallow fetch with default depth of 50, these arguments are used to override it
+# See here for more about that https://docs.gitlab.com/ee/ci/pipelines/settings.html#git-shallow-clone  # disable-secrets-detection
+  GIT_STRATEGY: clone
+  GIT_DEPTH: 0
+
+
+stages:
+  - unittests-and-validations
+  - create-instances
+  - run-instances
+  - upload-to-marketplace
```

Comment:
where are the `unittests-and-validations` and the `upload-to-marketplace` stages?

Reply:
Will be added by @DinaMeylakh